.github/workflows/run-coverage.yml

```diff
@@ -0,0 +1,60 @@
+name: run_coverage
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: Git ref on which to run the tests.
+        type: string
+        required: true
+  workflow_call:
+    inputs:
+      ref:
+        description: Git ref on which to run the tests.
+        type: string
+
+jobs:
+  coverage:
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.11"]
+        os: [ubuntu-latest]
+        extras: ["test-full"]
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout given ref
+      uses: actions/checkout@v3
+      if: inputs.ref != ''
+      with:
+        ref: ${{ inputs.ref }}
+
+    - name: Checkout current branch
+      uses: actions/checkout@v3
+      if: inputs.ref == ''
+      with:
+        ref: ${{ github.ref }}
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: 'pip'
+
+    - name: Update pip
+      run: |
+        python -m pip install --upgrade pip
+
+    - name: Install Scenic and dependencies
+      run: |
+        python -m pip install -e ".[${{ matrix.extras }}]"
+
+    - name: Run and report code coverage
+      run: |
+        coverage run -m pytest
```

Comment:
I think if you use `coverage` directly it doesn't capture subprocesses (in the coverage report I don't see any entry for `__main__.py`, presumably because all the tests for it use subprocesses). Probably better to use `pytest --cov`.

Reply:
Agreed, I did notice a slight coverage percentage difference between `pytest --cov` and `coverage` and reading up on the differences, I believe you are right on the ability to capture subprocesses. I will adjust accordingly!